src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -55,16 +57,24 @@ internal class ConnectionManager : IConnectionManager {
             interactiveWorkflow.RSession.Connected += RSessionOnConnected;
             interactiveWorkflow.RSession.Disconnected += RSessionOnDisconnected;
 
-            _shell.DispatchOnUIThread(() => _disposableBag.Add(_statusBar.AddItem(new ConnectionStatusBar { DataContext = _statusBarViewModel })));
+            _shell.Idle += OnIdle;
 
             // Get initial values
-            var userConnections = GetConnectionsFromSettings();
+            var userConnections = CreateConnectionList();
             _userConnections = new ConcurrentDictionary<Uri, IConnection>(userConnections);
 
             UpdateRecentConnections();
             SwitchBrokerToLastConnection();
         }
 
+        private void OnIdle(object sender, EventArgs e) {
+            _shell.Idle -= OnIdle;
+
+            _disposableBag.Add(_statusBar.AddItem(new ConnectionStatusBar {
```

Comment:
ok

---
In reply to: [77704011](https://github.com/Microsoft/RTVS/pull/2246#discussion_r77704011) [](ancestors = 77704011)